winpr/libwinpr/utils/ssl.c

```diff
@@ -244,9 +244,17 @@ static BOOL winpr_enable_fips(DWORD flags)
 #else
 		WLog_DBG(TAG, "Ensuring openssl fips mode is ENabled");
 
+#if (OPENSSL_VERSION_NUMBER > 0x10101000L)
```

Comment:
you might want to change the version to be `>1.1.2a`  instead of `1.1.1a`

Reply:
the 3.0 alpha versions did introduce `OPENSSL_VERSION_MAJOR` `OPENSSL_VERSION_MINOR` and `OPENSSL_VERSION_PATCH` so that might be a good way to fix the test

Reply:
So I have read some more documents (e.g. https://wiki.openssl.org/index.php/FIPS_modules) and my understanding is that FIPS mode was supported by 1.0.x versions, but it is not supported in 1.1.x versions at all  (although `FIPS_mode` and `FIPS_mode_set` exists) and going to be supported by 3.x. So `OPENSSL_VERSION_MAJOR >= 3` is probably the right way to go here...